dips/copy_to_netx.py

```diff
@@ -0,0 +1,298 @@
+
+
+def parse_component_id_from_mets(mets_path):
+    """Parse component ID from METS file
+
+    If a Dublin Core identifier is specified, it'll be used as a component ID.
+
+    Otherwise, if a netx.json file is added to the metadata folder of a
+    transfer, in the below format, then the value will end up in the METS file
+    and will be used as a component ID.
+
+    [{"component.identifier": "someidentifier"}]
+
+    This logic should be made more robust in the future.
+
+    """
+    mets = lxml.etree.parse(mets_path)
+
+    # First attempt to find it as the Dublin Core identifier
+    identifier = mets.find("//{http://purl.org/dc/elements/1.1/}identifier")
+
+    if identifier is not None:
+        return identifier.text
+
+    # Next attempt to find it as parsed from JSON
+    identifier = mets.find("//Componentidentifier")
+
+    if identifier is not None:
+        return identifier.text
+
+
+def parse_object_id_from_mets(mets_path):
+    """Parse object ID from METS file
+
+    If an accession number is specified when starting a transfer then the
+    value will end up in the METS file.
+
+    This logic should be made more robust in the future.
+
+    """
+    mets = lxml.etree.parse(mets_path)
+    accession_number = mets.find("//MetsMetsHdrAltRecordID")
+
+    if accession_number is not None:
+        return accession_number.text
+
+
+def write_csv_and_copy_objects(
+    netx_csv_directory, netx_objects_directory, dip_path, object_id, component_id
+):
+    """Cycle through objects, copying to destination path and and writing CSV file of files/attributes"""
+    csv_filepath = os.path.join(netx_csv_directory, "metadata.csv")
+    needs_header = False if os.path.isfile(csv_filepath) else True
+
+    with open(csv_filepath, "a") as csv_file:
+        writer = csv.writer(
+            csv_file, delimiter=",", quotechar="|", quoting=csv.QUOTE_MINIMAL
+        )
+
+        if needs_header:
+            # write header CSV row
+            writer.writerow(["filename", "ObjectID", "Component Number", "dip_uuid"])
+
+        objects_path = os.path.join(dip_path, "objects")
+        sip_uuid = uuid_from_dip_path(dip_path)
+
+        for object_file in os.listdir(objects_path):
+            # copy file to NetX directory
+            shutil.copyfile(
+                os.path.join(objects_path, object_file),
+                os.path.join(netx_objects_directory, object_file),
+            )
+
+            # write CSV row
+            writer.writerow([object_file, object_id, component_id, sip_uuid])
+
+
+def main(
+    shared_directory,
+    dip_path,
+    netx_csv_directory,
+    netx_objects_directory,
+    object_id,
+    component_id,
+    delete_local_copy,
+):
+    # Copy DIP to the currently processing location path, do not use any of the
+    # existing watched directories as that may trigger other workflows.
+    at_dips_dir = os.path.join(
+        shared_directory, "watchedDirectories", "automationToolsCopyToNetX"
+    )
```

Comment:
Is there a need to use the shared directory? I'd try to avoid that if possible, maybe using a working or output directory param.